src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,128 @@
+            if (resultSet.next()) {
+                String title = resultSet.getObject("title", String.class);
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+
+                Book book = new Book();
+                book.setId(id);
+                book.setPrice(price);
+                book.setTitle(title);
+                return Optional.of(book);
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book from DB",e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        List<Book> books = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String title = resultSet.getObject("title", String.class);
+                BigDecimal price = resultSet.getObject("price", BigDecimal.class);
+
+                Book book = new Book();
+                book.setId(id);
+                book.setPrice(price);
+                book.setTitle(title);
+                books.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find books from DB",e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            if (statement.executeUpdate() < 1) {
+                throw new RuntimeException("Expected to insert 1 record but inserted 0 rows");
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book from DB",e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        if (id == null) {
+            return false;
+        }
+        String sql = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            if (statement.executeUpdate() < 1) {
+                return false;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book from DB",e);
```

Comment:
The exception message should be more informative. Follow the good practice example from the checklist: 'throw new DataProcessingException("Can't delete book by id " + id, e);'.